src/twitter/twitter-notifications/twitternotificationsyncadaptor.cpp

```diff
@@ -0,0 +1,628 @@
+
+            qlonglong localId = (0 + notification->replacesId());
+            if (localId == 0) {
+                // failed.
+                SOCIALD_LOG_ERROR("failed to publish mention notification:" <<  body);
+            }
+        }
+    } else {
+        // error occurred during request.
+        SOCIALD_LOG_ERROR("unable to parse mention notification data from request with account" << accountId << "," <<
+                          "got:" << QString::fromLatin1(replyData.constData()));
+    }
+
+    // we're finished this request.  Decrement our busy semaphore.
+    decrementSemaphore(accountId);
+}
+
+void TwitterNotificationSyncAdaptor::finishedRetweetsHandler()
+{
+    QNetworkReply *reply = qobject_cast<QNetworkReply*>(sender());
+    int accountId = reply->property("accountId").toInt();
+
+    QByteArray replyData = reply->readAll();
+    disconnect(reply);
+    reply->deleteLater();
+    removeReplyTimeout(accountId, reply);
+
+    if (syncAborted()) {
+        SOCIALD_LOG_INFO("sync aborted, ignoring request response");
+        decrementSemaphore(accountId);
+        return;
+    }
+
+    bool ok = false;
+    QJsonArray tweets = parseJsonArrayReplyData(replyData, &ok);
+    if (ok) {
+        if (!tweets.size()) {
+            SOCIALD_LOG_DEBUG("no retweets received for account" << accountId);
+            decrementSemaphore(accountId);
+            return;
+        }
+
+        QString selfUserScreenName;
+        QHash<QString, int> dbRetweetCounts = m_db.retweetedTweetCounts(accountId);
+        QHash<QString, int> retweetCounts;
+        QStringList newlyRetweetedTweets;
+        int delta = 0;
+        foreach (const QJsonValue &tweetValue, tweets) {
+            QJsonObject tweet = tweetValue.toObject();
+            selfUserScreenName = tweet.value(QLatin1String("user")).toObject().value(QLatin1String("screen_name")).toString();
+            QString retweetId = tweet.value(QLatin1String("id_str")).toString();
+            int retweetsCount = tweet.value(QLatin1String("retweet_count")).toInt();
+            retweetCounts.insert(retweetId, retweetsCount);
+            if (!dbRetweetCounts.contains(retweetId) || dbRetweetCounts.value(retweetId) < retweetsCount) {
+                delta += retweetsCount - dbRetweetCounts.value(retweetId);
+                newlyRetweetedTweets.append(retweetId);
+            }
+        }
+
+        m_db.setRetweetedTweetCounts(accountId, retweetCounts); // won't get committed until finalize();
+        if (newlyRetweetedTweets.size() > 0) {
+            // Search if we already have a notification
+            Notification *notification = createNotification(accountId, Retweet);
+
+            // Set properties of the notification
+            QStringList openUrlArgs;
+            notification->setItemCount(newlyRetweetedTweets.size());
+            notification->setTimestamp(QDateTime::currentDateTimeUtc());
+            if (newlyRetweetedTweets.size() == 1) {
```

Comment:
This scheme doesn't work because the translation depends on both the number of your tweets and the number of retweets, which are split into two strings.